plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergColumnHandle.java

```diff
@@ -25,11 +25,26 @@
 import java.util.Objects;
 import java.util.Optional;
 
+import static io.trino.plugin.iceberg.ColumnIdentity.primitiveColumnIdentity;
+import static io.trino.spi.type.BigintType.BIGINT;
 import static java.util.Objects.requireNonNull;
+import static org.apache.iceberg.MetadataColumns.IS_DELETED;
+import static org.apache.iceberg.MetadataColumns.ROW_POSITION;
 
 public class IcebergColumnHandle
         implements ColumnHandle
 {
+    public static final IcebergColumnHandle ROW_POSITION_HANDLE = new IcebergColumnHandle(
+            primitiveColumnIdentity(ROW_POSITION.fieldId(), ROW_POSITION.name()),
+            BIGINT,
+            ImmutableList.of(),
+            BIGINT,
+            Optional.empty());
+
+    // use Integer.MIN_VALUE as $row_id field ID, which is currently not reserved by Iceberg
```

Comment:
would we know if iceberg starts using this id internally? Would sth break?

Reply:
Because Iceberg column name and type could change, comparing column ID is needed for equality check. Using a duplicated column ID would cause issue in that area.

All the Iceberg metadata columns have IDs counting down from `Integer.MAX_VALUE`. That's why I am choosing `Integer.MIN_VALUE` here. If there is any risk of conflicting IDs, Iceberg will inform the Trino community, and switching to another ID is fully backwards compatible.